### Beauty Ahmad

In [6]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [7]:
beauty_df = pd.read_csv("beauty.csv")


In [8]:
beauty_df.head()

Unnamed: 0  overall  verified  reviewTime      reviewerID        asin  \
0           0        5      True  2016-09-01  A3CIUOJXQ5VDQ2  B0000530HU   
1           1        5      True  2013-11-14  A3H7T87S984REU  B0000530HU   
2           2        1      True  2013-08-18  A3J034YH7UG4KT  B0000530HU   
3           3        5     False  2011-05-03  A2UEO5XR3598GI  B0000530HU   
4           4        5      True  2011-05-06  A3SFRT223XXWF7  B00006L9LC   

      reviewerName                                         reviewText  \
0         Shelly F                   As advertised. Reasonably priced   
1     houserules18  Like the oder and the feel when I put it on my...   
2             Adam  I bought this to smell nice after I shave.  Wh...   
3           Rich K  HEY!! I am an Aqua Velva Man and absolutely lo...   
4  C. C. Christian  If you ever want to feel pampered by a shampoo...   

          Size:  
0        7.0 oz  
1        7.0 oz  
2        7.0 oz  
3        7.0 oz  
4   200ml/6.7oz

In [9]:
del beauty_df["Unnamed: 0"]

In [10]:
beauty_df['reviewTime'] = pd.to_datetime(beauty_df['reviewTime'])
#beauty_df.head()

In [11]:
beauty_df.rename(columns={"overall":"productRating","asin":"productID", "Size:":"size"}, inplace=True)
beauty_df.head()

productRating  verified reviewTime      reviewerID   productID  \
0              5      True 2016-09-01  A3CIUOJXQ5VDQ2  B0000530HU   
1              5      True 2013-11-14  A3H7T87S984REU  B0000530HU   
2              1      True 2013-08-18  A3J034YH7UG4KT  B0000530HU   
3              5     False 2011-05-03  A2UEO5XR3598GI  B0000530HU   
4              5      True 2011-05-06  A3SFRT223XXWF7  B00006L9LC   

      reviewerName                                         reviewText  \
0         Shelly F                   As advertised. Reasonably priced   
1     houserules18  Like the oder and the feel when I put it on my...   
2             Adam  I bought this to smell nice after I shave.  Wh...   
3           Rich K  HEY!! I am an Aqua Velva Man and absolutely lo...   
4  C. C. Christian  If you ever want to feel pampered by a shampoo...   

           size  
0        7.0 oz  
1        7.0 oz  
2        7.0 oz  
3        7.0 oz  
4   200ml/6.7oz

In [12]:
beauty_df.groupby('productID').count().sort_values('productRating', ascending = False)

productRating  verified  reviewTime  reviewerID  reviewerName  \
productID                                                                   
B0012Y0ZG2           1794      1794        1794        1794          1794   
B000URXP6E            858       858         858         858           858   
B001OHV1H4            489       489         489         489           489   
B00006L9LC            489       489         489         489           489   
B0009RF9DW            392       392         392         392           392   
...                   ...       ...         ...         ...           ...   
B00MGK9Z8U              1         1           1           1             1   
B0091OCA86              1         1           1           1             1   
B00B7V273E              1         1           1           1             1   
B00GHJOM2U              1         1           1           1             1   
B01DKQAXC0              1         1           1           1             1   

            reviewText  size  
productID                     
B0012Y0ZG2        1792  1794  
B000URXP6E         857   857  
B001OHV1H4         488   489  
B00006L9LC         488   489  
B0009RF9DW         392   392  
...                ...   ...  
B00MGK9Z8U           1     0  
B0091OCA86           1     0  
B00B7V273E           1     1  
B00GHJOM2U           1     0  
B01DKQAXC0           1     0  

[85 rows x 7 columns]

#### The Data at a glance

In [13]:
beauty_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5269 entries, 0 to 5268
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   productRating  5269 non-null   int64         
 1   verified       5269 non-null   bool          
 2   reviewTime     5269 non-null   datetime64[ns]
 3   reviewerID     5269 non-null   object        
 4   productID      5269 non-null   object        
 5   reviewerName   5269 non-null   object        
 6   reviewText     5264 non-null   object        
 7   size           4358 non-null   object        
dtypes: bool(1), datetime64[ns](1), int64(1), object(5)
memory usage: 293.4+ KB


In [14]:
beauty_df.isnull().sum()

productRating      0
verified           0
reviewTime         0
reviewerID         0
productID          0
reviewerName       0
reviewText         5
size             911
dtype: int64

 I renamed the columns to clearer terms, I also figured I have 911 null values in size

#### Analysis with datetimes

In [15]:
time_span = beauty_df["reviewTime"].max() - beauty_df["reviewTime"].min()

print ("Time span of the data: {}".format(time_span))

Time span of the data: 5084 days 00:00:00


In [16]:
beauty_df["year"] = beauty_df["reviewTime"].dt.year
beauty_df["month"] = beauty_df["reviewTime"].dt.month_name()
beauty_df["dayOfTheWeek"] = beauty_df["reviewTime"].dt.day_name()

In [23]:
year_count = beauty_df.groupby("year").count().sort_values(
    "reviewText", ascending=False)[["reviewText"]]
year_count.rename(columns={"reviewText":"numberOfReviews"}, inplace=True)

print("AMOUNT OF REVIEWS PER YEAR:")
year_count

AMOUNT OF REVIEWS PER YEAR:


numberOfReviews
year                 
2016             1232
2015             1027
2017              825
2014              784
2013              756
2018              443
2012              111
2009               25
2008               20
2011               20
2007               13
2010                4
2006                2
2004                1
2005                1

In [24]:
month_count = beauty_df.groupby("month").count().sort_values(
    "reviewText", ascending=False)[["reviewText"]]
month_count.rename(columns={"reviewText":"numberOfReviews"}, inplace=True)

print("AMOUNT OF REVIEWS PER MONTH:")
month_count

AMOUNT OF REVIEWS PER MONTH:


numberOfReviews
month                     
August                 598
September              550
April                  548
March                  518
July                   464
May                    445
February               413
January                410
December               352
November               338
October                320
June                   308

In [25]:
year_month_count = beauty_df.groupby(["year", "month"]).count().sort_values(
    "year", ascending=False)[["reviewText"]]
year_month_count.rename(columns={"reviewText":"numberOfReviews"}, inplace=True)

print("MONTHLY BREAKDOWN OF AMOUNT OF REVIEWS FOR EACH YEAR:")
year_month_count

MONTHLY BREAKDOWN OF AMOUNT OF REVIEWS FOR EACH YEAR:


numberOfReviews
year month                     
2018 May                    102
     March                   77
     June                     3
     July                     1
     January                 15
...                         ...
2007 December                 8
     April                    2
2006 May                      2
2005 June                     1
2004 September                1

[101 rows x 1 columns]

In [26]:
yearly_rating = beauty_df.groupby("year").agg({"reviewText":"count","productRating":"mean"}).sort_values(
    "reviewText",ascending=False)
yearly_rating.rename(columns={"reviewText":"numberOfReviews", "productRating":"aveargeProductRating"}, inplace=True)

print("AVERAGE RATINGS PER YEAR:")
yearly_rating

AVERAGE RATINGS PER YEAR:


numberOfReviews  aveargeProductRating
year                                       
2016             1232              4.933442
2015             1027              4.932171
2017              825              4.713939
2014              784              4.891582
2013              756              4.880952
2018              443              3.724605
2012              111              4.927928
2009               25              4.680000
2008               20              3.200000
2011               20              4.400000
2007               13              4.230769
2010                4              5.000000
2006                2              5.000000
2004                1              5.000000
2005                1              5.000000

In [27]:
yearly_month_rating = beauty_df.groupby(["year","month"]).agg({"reviewText":"count", "productRating":"mean"}).sort_values(
    "year", ascending=False)
yearly_month_rating.rename(columns={"reviewText":"numberOfReviews", "productRating":"aveargeProductRating"}, inplace=True)

print("MONTHLY BREAKDOWN FOR AVERAGE RATINGS PER YEAR:")
yearly_month_rating

MONTHLY BREAKDOWN FOR AVERAGE RATINGS PER YEAR:


numberOfReviews  aveargeProductRating
year month                                           
2018 May                    102              3.549020
     March                   77              2.298701
     June                     3              4.333333
     July                     1              5.000000
     January                 15              5.000000
...                         ...                   ...
2007 December                 8              4.250000
     April                    2              5.000000
2006 May                      2              5.000000
2005 June                     1              5.000000
2004 September                1              5.000000

[101 rows x 2 columns]

### Product analysis


In [28]:
no_of_products = len(beauty_df["productID"].unique())
print("Amount of products that were reviewed: {}".format(no_of_products))

Amount of products that were reviewed: 85


In [29]:
reviews_per_product = beauty_df.groupby("productID").count().sort_values("reviewText", ascending=False)[["reviewText"]]
reviews_per_product.rename(columns={"reviewText":"numberOfReviews"}, inplace=True)

print("NUMBER OF REVIEWS PER PRODUCT: ")
reviews_per_product

NUMBER OF REVIEWS PER PRODUCT: 


numberOfReviews
productID                  
B0012Y0ZG2             1792
B000URXP6E              857
B001OHV1H4              488
B00006L9LC              488
B0009RF9DW              392
...                     ...
B00MGK9Z8U                1
B0091OCA86                1
B00B7V273E                1
B00GHJOM2U                1
B01DKQAXC0                1

[85 rows x 1 columns]

In [30]:
top_products = reviews_per_product["numberOfReviews"].nlargest(10)

print("MOST REVIEWED PRODUCTS: \n{}".format(top_products))

MOST REVIEWED PRODUCTS: 
productID
B0012Y0ZG2    1792
B000URXP6E     857
B001OHV1H4     488
B00006L9LC     488
B0009RF9DW     392
B000FI4S1E     392
B00W259T7G     163
B0010ZBORW      98
B0013NB7DW      40
B001LNODUS      33
Name: numberOfReviews, dtype: int64


In [31]:
least_products = reviews_per_product["numberOfReviews"].nsmallest(10)

print("LEAST REVIEWED PRODUCTS: \n{}".format(least_products))

LEAST REVIEWED PRODUCTS: 
productID
B00MGK9Z8U    1
B0091OCA86    1
B00B7V273E    1
B00GHJOM2U    1
B01DKQAXC0    1
B00126LYJM    2
B0002JHI1I    2
B0014SQQ3M    2
B00155Z6V2    2
B00157OBRU    2
Name: numberOfReviews, dtype: int64
